kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -372,4 +373,294 @@ void AuxiliarModelPartUtilities::RemoveConditionsAndBelongingsFromAllLevels(Flag
     aux_utility.RemoveConditionsAndBelongings(IdentifierFlag);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+ModelPart& AuxiliarModelPartUtilities::DeepCopyModelPart(
+    const std::string& rNewModelPartName,
+    Model* pModel
+    )
+{
+    // First we check if the model is not null
+    if (pModel == nullptr)
+        pModel = &mrModelPart.GetModel();
+
+    // We create the new model part
+    ModelPart& r_model_part = pModel->CreateModelPart(rNewModelPartName);
+
+    /// We will copy the member variables of the model part one by one
+
+    // We copy the buffer size (direct copy)
+    r_model_part.SetBufferSize(mrModelPart.GetBufferSize());
+
+    // We copy the process info
+    r_model_part.GetProcessInfo() = mrModelPart.GetProcessInfo();
+
+    // We copy the list of variables (using copy constructor)
+    r_model_part.SetNodalSolutionStepVariablesList(Kratos::make_intrusive<VariablesList>(mrModelPart.GetNodalSolutionStepVariablesList()));
+    r_model_part.SetNodalSolutionStepVariablesList();
+
+    // We copy the tables, first using the copy constructor, and then reassigning each table so it doesn't point to the original one
+    const auto& r_reference_tables = mrModelPart.Tables();
```

Comment:
I would create a `PointerVectorSetDeepCopy` method and would use it for all the entities

Reply:
Makes sense, better public so can be used outside this class

Reply:
There is a problem, there is no consistency between constructors and Clone methods, the only thing I can easily unify are conditions and elements as they have the same interface.

Reply:
I did some refactor with Condition and Elements, the other classes are quite complex due to very different constructors and Clone methods